### ISSUE TITLE
Need better (any) signal handling during cleanup

### ISSUE THREAD

**Issue by richarddli** [telepresenceio/telepresence#358]
I do:

```
$ telepresence --namespace dashboard --swap-deployment issues-master --docker-run --rm -i -t -v $(pwd):/service issues-dev
```

This works great. Then I Ctrl-C out of the process.

```
lost-cause:issues richard$ kubectl get deploy -n dashboard
NAME             DESIRED   CURRENT   UP-TO-DATE   AVAILABLE   AGE
issues-master    0         0         0            0           1m
```

The deployment is broken, and I have to recreate the deployment object.

**Comment by ark3** [comment]
Richard sent me this additional snippet:

```
[...]
127.0.0.1 - - [30/Nov/2017 18:45:53] "GET / HTTP/1.1" 200 -
127.0.0.1 - - [30/Nov/2017 18:45:54] "GET / HTTP/1.1" 200 -
^C^CError in atexit._run_exitfuncs:
Traceback (most recent call last):
  File "/usr/local/Cellar/python3/3.6.3/Frameworks/Python.framework/Versions/3.6/lib/python3.6/subprocess.py", line 1457, in wait
    (pid, sts) = self._try_wait(0)
  File "/usr/local/Cellar/python3/3.6.3/Frameworks/Python.framework/Versions/3.6/lib/python3.6/subprocess.py", line 1404, in _try_wait
    (pid, sts) = os.waitpid(self.pid, wait_flags)
KeyboardInterrupt
```

It seems there are two things going wrong here. First, Richard's Ctrl-C is being sent to Telepresence (yielding the KeyboardInterrupt traceback _from Telepresence_) instead of just to the container process. Second, because Telepresence is crashing (due to the KeyboardInterrupt) during exit, we run into #260, i.e. the cluster is not being cleaned up.

The quick workaround for Richard is to add `--init` to the `--docker-run` arguments, e.g., 

```
$ telepresence --namespace dashboard --swap-deployment issues-master --docker-run --init --rm -i -t -v $(pwd):/service issues-dev
```

This causes docker to "Run an init inside the container that forwards signals and reaps processes." The net effect is that Ctrl-C exits the container as intended and Telepresence exits normally, thus cleaning up the cluster.

Note that **Telepresence is supposed to do this automatically** in `run_docker_command(...)`. We should figure out why that did not happen.

**Comment by ark3** [comment]
There are three problems here.

1. Pressing Ctrl-C exits the container and Telepresence starts cleanup. However, it does not tell the user this, so it appears as if nothing has happened.
2. Pressing Ctrl-C _again_ kills Telepresence during the cleanup process, i.e. Telepresence crashes with a `KeyboardInterrupt` exception. Telepresence ought to be resistant to Ctrl-C killing it during cleanup.
3. Because the user has killed Telepresence, cleanup does not occur, so we experience issue #260.

This particular issue should address item 2 and possibly item 1.

The `docker --init` stuff mentioned above is a distraction; Telepresence _does_ add that automatically.

**Comment by ark3** [comment]
Given that item 1 above is covered by #594, let's make this issue cover item 2, i.e. make this the "better signal handling during cleanup" issue.

**Comment by ark3** [comment]
I suspect that once we fix #586, this will be resolved.

**Comment by donnyyung** [comment]
I believe this is no longer an issue in Telepresence 2. Here are the docs on how to install Telepresence (https://www.telepresence.io/docs/latest/install/), please re-open if you still see this issue in our latest version!
